packages/@aws-cdk/core/lib/fs/options.ts

```diff
@@ -85,9 +75,30 @@ export interface CopyOptions {
 }
 
 /**
- * Options related to calculating source hash.
+ * Obtains applied when copying directories into the staging location.
+ */
+export interface CopyOptions extends FileOptions {
+  /**
+   * A strategy for how to handle symlinks.
+   *
+   * @default SymlinkFollowMode.NEVER
+   */
+  readonly follow?: SymlinkFollowMode;
+}
+
+/**
+ * Options applied when copying directories into the staging location.
  */
-export interface FingerprintOptions extends CopyOptions {
+export interface FileCopyOptions extends FileOptions {
```

Comment:
Do we need both of these or can we deprecate the former?

Reply:
Again, sorry, I don't understand what you're asking here (deprecate which interface?).

Reply:
You don't have to be sorry. It's ok.

I was asking if we should  `CopyOptions` and `FileCopyOptions`. They look exactly the same.

Reply:
Sure, we can deprecate `CopyOptions`. It's used in a bunch of public classes, so it will require changes to augment `CopyOptions` with `FileCopyOptions` in all of these usages, and it's not linked to deprecating `@aws-cdk/assets` directly.

Let me know if you want to make this part of this PR, or tackle it separately after this is merged.

Reply:
Can you get @eladb's eyes on this? He owns and has built this area and it'd best for him to take a look at this PR and assess this change.

I'll defer to him on what the right thing here is.